mindsdb/interfaces/agents/agents_controller.py

```diff
@@ -223,72 +228,56 @@ def add_agent(
             # It will be picked up by get_api_key() in handler_utils.py
             pass
 
-        # Extract table and knowledge base parameters from params
-        database = params.pop("database", None)
-        knowledge_base_database = params.pop("knowledge_base_database", DEFAULT_TEXT2SQL_DATABASE)
-        include_tables = params.pop("include_tables", None)
-        ignore_tables = params.pop("ignore_tables", None)
-        include_knowledge_bases = params.pop("include_knowledge_bases", None)
-        ignore_knowledge_bases = params.pop("ignore_knowledge_bases", None)
-
-        # Save the extracted parameters back to params for API responses only if they were explicitly provided
-        # or if they're needed for skills
-        need_params = include_tables or ignore_tables or include_knowledge_bases or ignore_knowledge_bases
-
-        if "database" in params or need_params:
-            params["database"] = database
+        depreciated_params = [
+            "database",
+            "knowledge_base_database",
+            "include_tables",
+            "ignore_tables",
+            "include_knowledge_bases",
+            "ignore_knowledge_bases",
+        ]
+        if any(param in params for param in depreciated_params):
+            raise ValueError(
```

Comment:
it makes the rest api not back compatible, do the customers use these parameters? 
and also it has to be checked if they are used by minds or other mindsdb components